web/styles/rendered_markdown.css

```diff
@@ -23,15 +23,39 @@
         margin-left: 0;
     }
 
-    /* Ensure ordered lists are nicely centered in 1-line messages */
     & ol {
-        margin: 2px 0 5px 20px;
+        margin: 2px 0 5px;
+
+        /* Ensure ordered lists are nicely centered in 1-line messages for up to 2
+           digit markers, else the markers take up as much space as needed without
+           being cut off at the beginning */
+        > li {
+            display: grid;
+            grid-template-columns: minmax(20px, max-content) 1fr;
```

Comment:
So this `1fr` value is a problem, as it invites a grid blowout on something long and unbreakable, like this string of nonsense alpha text:

<img width="865" alt="Screenshot 2023-09-08 at 1 22 29 PM" src="https://github.com/zulip/zulip/assets/170719/577b0cf1-aa3d-42f8-856c-3f6450997518">

Essentially, `1fr` is equivalent to `minmax(max-content, 1fr)`--so what needs to happen here is for you to explicitly say that `0` is the minimum acceptable value--`minmax(0, 1fr)`, which allows even long text to wrap, and for the numbers to be undisturbed:

<img width="865" alt="Screenshot 2023-09-08 at 1 22 54 PM" src="https://github.com/zulip/zulip/assets/170719/b5aaa901-fd6b-4e8f-8ddb-bd9ad2be9227">

You can read a bit more behind the theory of `minmax(0, 1fr)` in #26615.

Reply:
This was new to me, interesting how `fr` works... thanks for the clear explanation and context!

Fixed it:
![image](https://github.com/zulip/zulip/assets/68962290/f2af0397-f6cf-4a06-ab28-89a22e9b4227)